server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1987,6 +2058,15 @@ private SnapshotsInProgress updatedSnapshotsInProgress(ClusterState currentState
             }
             return changed ? SnapshotsInProgress.of(snapshotEntries) : null;
         }
+
+        private void markShardReassigned(String indexName, int shardId, Map<String, Set<Integer>> reassignments) {
+            final boolean added = reassignments.computeIfAbsent(indexName, k -> new HashSet<>()).add(shardId);
+            assert added;
```

Comment:
Maybe add some message in case it occurs in tests?

Reply:
++